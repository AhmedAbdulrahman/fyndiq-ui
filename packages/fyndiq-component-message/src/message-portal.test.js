import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import MessagePortal from './message-portal'

describe('fyndiq-component-message MessagePortal', () => {
  it('should return a Portal', () => {
    // Mock the Portal behavior to return its content
    ReactDOM.createPortal = jest.fn(children => children)
    const Component = shallow(<MessagePortal>Content</MessagePortal>)
    expect(Component).toMatchSnapshot()
  })

  it('should create a portal div on mount', () => {
    shallow(<MessagePortal portalId="test-id" />)
    expect(document.getElementById('test-id')).not.toBe(null)
  })

  it('should not create another portal with the same ID', () => {
    shallow(<MessagePortal portalId="test-id" />)
    shallow(<MessagePortal portalId="test-id" />)
    expect(document.querySelectorAll('#test-id')).toHaveLength(1)
  })
})
