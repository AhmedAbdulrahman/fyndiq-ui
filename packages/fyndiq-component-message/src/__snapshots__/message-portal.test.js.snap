// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-message MessagePortal should return a Portal 1`] = `"Content"`;
